script.js

```diff
@@ -1,100 +1,332 @@
+formGastoFijo.addEventListener("submit", function (e) {
+  e.preventDefault();
+  const concepto = document.getElementById("concepto-fijo").value.trim();
+  const costo = parseFloat(document.getElementById("costo-fijo").value);
+
+  if (concepto && !isNaN(costo)) {
     calculadora.agregarGastoFijo(concepto, costo);
-     alert(" Gasto fijo agregado correctamente.");
-    }
-    seguirGasto = confirm("¿Quiere agregar otro gasto?");
+  actualizarTablaGastosFijos();
+    guardarDatosEnStorage();
+
+    formGastoFijo.reset();
+
+     // Mostrar modal manualmente
+  const modalGasto = new bootstrap.Modal(document.getElementById('agregarGasto'));
+  modalGasto.show();
   }
-  break;
+});
 
-case "3":
-  gananciaPorcentual = parseFloat(prompt("Ingrese el porcentaje de ganancia que desee (ej: 25%):"));
-   cantidadProduccion = parseInt(prompt("Ingrese la cantidad total a producir (ej: 200):"));
+// Evento para calcular costos totales de producción
+// se aplivca ganancia porcentual y muestra resultados 
+formCalculadora.addEventListener("submit", function (e) {
+  e.preventDefault();
 
-  if (isNaN(cantidadProduccion) || isNaN(gananciaPorcentual) || cantidadProduccion <= 0) {
-    alert("Cantidad o ganancia ingresadas inválidas.");
-  }else{
-    alert("Valores guardados correctamente")
+  const gananciaPorcentual = parseFloat(document.getElementById("ganancia").value);
+  const cantidadProduccion = parseInt(document.getElementById("cantidadTotal").value);
+
+  // se valida si hay al menos un material o gasto fijo
+  if (calculadora.materiales.length === 0 && calculadora.gastosFijos.length === 0) {
+    const modalAntesDeCalcular = new bootstrap.Modal(document.getElementById('modalAntesDeCalcular'));
+    modalAntesDeCalcular.show();
+    return;
+  }
+
+  // se validan datos de ganancia y cantidad de producción
+  if (!isNaN(gananciaPorcentual) && !isNaN(cantidadProduccion) && cantidadProduccion > 0) {
+    const resumen = calculadora.generarResumen(gananciaPorcentual, cantidadProduccion);
+
+    const totalVariables = calculadora.calcularCostoMaterialesTotal(cantidadProduccion);
+    const totalFijos = calculadora.calcularGastosFijos();
+    const costoTotal = totalVariables + totalFijos;
+    const costoTotalConGanancia = costoTotal * (1 + gananciaPorcentual / 100);
+    const costoUnidad = costoTotalConGanancia / cantidadProduccion;
+
+    totalVariablesDOM.textContent = `$${totalVariables.toFixed(2)}`;
+    totalFijosDOM.textContent = `$${totalFijos.toFixed(2)}`;
+    costoTotalDOM.textContent = `$${costoTotalConGanancia.toFixed(2)}`;
+    costoGananciaDOM.textContent = `$${costoUnidad.toFixed(2)}`;
+
+    const modalFinal = new bootstrap.Modal(document.getElementById('calculosFinales'));
+    modalFinal.show();
+  }
+});
+
+// Vacía completamente la lista de materiales después de una confirmación del usuario
+document.getElementById("vaciar-materiales").addEventListener("click", function () {
+  if (calculadora.materiales.length === 0) {
```

Comment:
Veo que usas mucho esta llamada, ¿no se puede abstraer a Calculadora?
```suggestion
calculadora.isMaterialsEmpty()
```